content/posts/2021-11-21-tăng-tốc.md

```diff
@@ -0,0 +1,65 @@
+
+### 5. License
+![License](/media/license.jpg)
+
+It's very important that the package you install grant you permission to use it for commercial in production. Packages in the wild are great. But not all of them are free to use. 
+
+Let's take an example. If you are working with charts, you might hear of [Highcharts](https://github.com/highcharts/highcharts) - Rock-solid and incredibly flexible charting library made for developers. You download it from [npm](https://www.npmjs.com/), make a chart, it looks great, and you are ready to roll out the feature. But please note that Highcharts is not free for commercial. So if you are writing code for your company, haven't bought a license and use this package for production, you are basically breaking the law. We don't want that, right? So please buy a license before using Highcharts in production. Myself, I use [Recharts](https://github.com/recharts/recharts) instead, a free package powered by [D3](https://github.com/d3/d3) with MIT license and in very active development.
+
+Software licenses are quite complicated and go beyond the purpose of this post. But in general, package with MIT, Apache 2.0 and BSD are safe to use for commercial. If you choose a package that's are not one of those, please please read the license carefully.
+
+### 6. Reasonable size
+![Size](/media/size.jpg)
+
+You don't want to bloat your web application's bundle size by installing a 700kb package. Sometimes you need to, but most of the time, you don't. Please consider the size of the library before making a decision. You need to know about the unpack size, the bundle size, the gzipped size. A larger package means you need more hard disk and bandwidth, which means more money you need to pay, also it takes a longer time for your users to use your app.
+
+### 7. DIY
+![Do it yourself](/media/diy.jpg)
+
+As mentioned in [Fit your need](#fit-your-need), many packages try to solve a problem in general (Thus the size of the package is large), you only need a small part of the package. Sometimes, your problem is unique and there are no existing third party packages out there to solve yours. In those cases, it's a great time for you to do it yourself. I find myself in the early days in the industry spending many times finding a third party package to help me build features. But over time, I rarely use external packages for my daily task. It doesn't mean that I always reinvent the wheel. It means that I know what I am doing and I can seek help from the community when I truly need to (for example I will never sanitize user input by myself but use [DOMPurify](https://github.com/cure53/DOMPurify))
```

Comment:
- Thus => thus. Đoạn này không viết hoa.
- in the industry => in this industry
- for my daily task => for my daily tasks